bin/rucio

```diff
@@ -1077,6 +1077,33 @@ def download(args):
             logger.debug(args.dids)
         item_defaults['pfn'] = args.pfn
         item_defaults['did'] = did_str
+        if args.rses is None:
+            logger.warning("No RSE was given, selecting one.")
+            client = get_client(args)
+            replicas = client.list_replicas(
+                [{"scope": did_str.split(':')[0], "name": did_str.split(':')[-1]}],
+                schemes=args.protocol,
+                ignore_availability=False,
+                client_location=detect_client_location(),
+                resolve_archives=not args.no_resolve_archives
+            )
+            try:
+                # Check each rse in the replica list for the pfn. If no pfn is found, raise StopIteration.
+                # If it is found, stop the generator and add the rse to 'item_defaults'
+                download_rse = next(
+                    next(
+                        rse for rse in replica['rses']
+                        if args.pfn in replica['rses'][rse]
+                    )
+                    for replica in replicas
+                )
```

Comment:
My understanding here is that, if the inner `next` exhausts the iterator, it raises a `StopIteration`, so the outer loop will never go beyond the first iteration.

For readability, I think it would be best to just use `for` loops and either:
1. refactor the loop into a function (e.g. `get_rse_by_pfn` or something along those lines) and `return` the rse if found, no return otherwise
2. keep the loop here and using `break` when a matching rse is found